### PR TITLE
FoodAdvisor V2

### PR DESCRIPTION
**Pull request by Mcastres**
This PR is about improving FoodAdvisor.

### CODE REVIEW

File: api/extensions/content-manager/admin/src/containers/EditView/index.js [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+                          icon="layout"
+                          url={configurationsURL}
+                          onClick={() => {
+                            // emitEvent('willEditContentTypeLayoutFromEditView');
+                          }}
+                        />
+                      </CheckPermissions>
+                      {!data.published_at && _.get(data, 'slug') && (
+                        <li>
+                          <StyledExternalLink
```
[the remark this answers is not in the document]
Comment (soupette) [reply api/extensions/content-manager/admin/src/containers/EditView/index.js:238]: Todo so:

1. Create extensions/content-manager/admin/src/InjectedComponents/ExternalLink/index.js

```js
import React from 'react';
import styled from 'styled-components';
import { EyeIcon, useContentManagerEditViewDataManager } from 'strapi-helper-plugin';

const StyledExternalLink = styled.a`
  display: block;
  color: #333740;
  width: 100%;
  text-decoration: none;
  span,
  i,
  svg {
    color: #333740;
    width: 13px;
    height: 12px;
    margin-right: 10px;
    vertical-align: 0;
  }
  span {
    font-size: 13px;
  }
  i {
    display: inline-block;
    background-image: url(${EyeIcon});
    background-size: contain;
  }
  &:hover {
    text-decoration: none;
    span,
    i,
    svg {
      color: #007eff;
    }
  }
`;

const ExternalLink = () => {
  const { modifiedData, layout } = useContentManagerEditViewDataManager();

  if (modifiedData.published_at) {
    return null;
  }

  if (!modifiedData.slug) {
    return null;
  }

  if (!CLIENT_URL || !CLIENT_PREVIEW_SECRET) {
    return null;
  }

  console.log(CLIENT_URL);

  return (
    <li>
      <StyledExternalLink
        href={`${CLIENT_URL}/api/preview?secret=${CLIENT_PREVIEW_SECRET}&slug=${modifiedData.slug}&locale=${modifiedData.locale}&apiID=${layout.apiID}&kind=${layout.kind}`}
        target="_blank"
        rel="noopener noreferrer"
        title="page preview"
      >
        <i />
        Preview
      </StyledExternalLink>
    </li>
  );
};

export default ExternalLink;

```

2. We will override (temporarily) du content-manager/admin/src/index.js. file until the v4

```js
// NOTE TO PLUGINS DEVELOPERS:
// If you modify this file by adding new options to the plugin entry point
// Here's the file: strapi/docs/3.0.0-beta.x/plugin-development/frontend-field-api.md
// Here's the file: strapi/docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md
// Also the strapi-generate-plugins/files/admin/src/index.js needs to be updated
// IF THE DOC IS NOT UPDATED THE PULL REQUEST WILL NOT BE MERGED

import pluginPkg from '../../package.json';
import pluginId from './pluginId';
import pluginLogo from './assets/images/logo.svg';
import App from './containers/Main';
import ExternalLink from './InjectedComponents/ExternalLink';
import ConfigureViewButton from './InjectedComponents/ContentTypeBuilder/ConfigureViewButton';
import lifecycles from './lifecycles';
import reducers from './reducers';
import trads from './translations';

export default (strapi) => {
  const pluginDescription = pluginPkg.strapi.description || pluginPkg.description;
  const plugin = {
    blockerComponent: null,
    blockerComponentProps: {},
    description: pluginDescription,
    icon: pluginPkg.strapi.icon,
    id: pluginId,
    initializer: null,
    injectedComponents: [
      {
        plugin: 'content-type-builder.listView',
        area: 'list.link',
        component: ConfigureViewButton,
        key: 'content-manager.link',
      },
      // This is the injection zone
      {
        plugin: 'content-manager.editView',
        area: 'right.links',
        component: ExternalLink,
        key: 'content-manager.preview-link',
      },
    ],
    injectionZones: {
      editView: { informations: [] },
      listView: { actions: [], deleteModalAdditionalInfos: [] },
    },
    isReady: true,
    isRequired: pluginPkg.strapi.required || false,
    layout: null,
    lifecycles,
    mainComponent: App,
    name: pluginPkg.strapi.name,
    pluginLogo,
    preventComponentRendering: false,
    reducers,
    trads,
  };

  return strapi.registerPlugin(plugin);
};

```

[the remark this answers is not in the document]
Comment (soupette) [reply api/extensions/content-manager/admin/src/containers/EditView/index.js:238]: Then you can remove this file completely

File: client/README.md [hunk 2]
```diff
@@ -1,61 +1,23 @@
-This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+# Next.js + Tailwind CSS Example
```
Comment (soupette) [inline client/README.md:1]: I think that you should change the README file

**Review by soupette — CHANGES_REQUESTED** [review]

File: api/extensions/content-manager/admin/src/containers/EditView/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mcastres) [reply api/extensions/content-manager/admin/src/containers/EditView/index.js:238]: Thanks it works well like this. However are you sure the `strapi-helper-plugin` has an `EyeIcon` svg?

[the remark this answers is not in the document]
Comment (soupette) [reply api/extensions/content-manager/admin/src/containers/EditView/index.js:238]: YEs it does

**Comment by fdel-car** [comment]
Hello, just listing some issues that I encountered during my test of the project itself:

- Our old version of sqlite3 has issues if only python3 is installed, which is the case for recent distributions. This could be bothering our users or even stop them completely in their tracks.
- The `seed.js` script in the `api` folder, is replacing everything inside of the `.env` file. Could be an issue as well when people follow the README in order.
- The Swagger plugin does not seem to be installed by default even though the guided tour says so.

 I'll update this list if I'm mistaken or if I find others and I'll start reviewing the PR itself soon I promise xD

**Comment by Mcastres** [comment]
Thanks for your comments Fabien! Everything has been fixed 👍

**Pull request merged**